include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -979,6 +1033,270 @@ namespace internal
+    {
+    public:
+      MGDataEntry(const DoFHandlerType *           dofs,
+                  const MGLevelObject<VectorType> *vectors,
+                  const std::vector<std::string> & names,
+                  const std::vector<
+                    DataComponentInterpretation::DataComponentInterpretation>
+                    &data_component_interpretation)
+        : DataEntryBase<DoFHandlerType>(dofs,
+                                        names,
+                                        data_component_interpretation)
+        , vectors(vectors)
+      {}
+
+      double // TODO
+      get_cell_data_value(const unsigned int /*cell_level*/,
+                          const unsigned int cell_index) const
+      {
+        Assert(false, ExcNotImplemented());
+
+        // cell_number is counting over all cells so compute the level and
+        // index from it
+        unsigned int idx = cell_index;
+        unsigned int lvl = vectors->min_level();
+        while ((*vectors)[lvl].size() <= idx)
+          {
+            idx -= (*vectors)[lvl].size();
+            ++lvl;
+          }
+        return get_vector_element((*vectors)[lvl], idx);
+      }
+
+
+      virtual double
+      get_cell_data_value(
+        const unsigned int       cell_number,
+        const ComponentExtractor extract_component) const override
+      {
+        Assert(false, ExcNotImplemented());
+
+        (void)cell_number;
+        (void)extract_component;
+        return 0.0;
+      }
+
+
+      virtual void
+      get_function_values(
+        const FEValuesBase<DoFHandlerType::dimension,
+                           DoFHandlerType::space_dimension> &fe_patch_values,
+        const ComponentExtractor                             extract_component,
+        std::vector<double> &patch_values) const override;
+
+      /**
+       * Given a FEValuesBase object, extract the values on the present cell
+       * from the vector we actually store. This function does the same as the
+       * one above but for vector-valued finite elements.
+       */
+      virtual void
+      get_function_values(
+        const FEValuesBase<DoFHandlerType::dimension,
+                           DoFHandlerType::space_dimension> &fe_patch_values,
+        const ComponentExtractor                             extract_component,
+        std::vector<dealii::Vector<double>> &patch_values_system)
+        const override;
+
+      /**
+       * Given a FEValuesBase object, extract the gradients on the present
+       * cell from the vector we actually store.
+       */
+      virtual void
+      get_function_gradients(
+        const FEValuesBase<DoFHandlerType::dimension,
+                           DoFHandlerType::space_dimension>
+          & /*fe_patch_values*/,
+        const ComponentExtractor /*extract_component*/,
+        std::vector<Tensor<1, DoFHandlerType::space_dimension>>
+          & /*patch_gradients*/) const override
+      {
+        Assert(false, ExcNotImplemented());
```

Comment:
I guess that means that most `DataPostprocessor`s aren't going to work with MG vectors, right?

Reply:
Correct. I have not tried them at all, so maybe none of them work correctly. This is something for a later time, I would think.